public/app/features/correlations/Forms/QueryEditorField.tsx

```diff
@@ -52,8 +124,31 @@ export const QueryEditorField = ({ dsUid, invalid, error, name }: Props) => {
           if (!QueryEditor) {
             return <Alert title="Data source does not export a query editor."></Alert>;
           }
-
-          return <QueryEditor onRunQuery={() => {}} onChange={onChange} datasource={datasource} query={value} />;
+          return (
+            <>
+              <QueryEditor
+                onRunQuery={() => handleValidation(value)}
+                onChange={(value) => {
+                  setIsValidQuery(undefined);
+                  onChange(value);
+                }}
+                datasource={datasource}
+                query={value}
+              />
+              <HorizontalGroup justify="flex-end">
+                {isValidQuery ? (
+                  <div className={style.valid}>
+                    <Icon name="check" /> This query is valid.
+                  </div>
+                ) : isValidQuery === false ? (
+                  <FieldValidationMessage>This query is not valid.</FieldValidationMessage>
+                ) : null}
+                <Button variant="primary" icon={'check'} type="button" onClick={() => handleValidation(value)}>
```

Comment:
Would it make sense to make it a `secondary` button? It's optional action and being close to the "Save" button looks like it is as important as saving.

Reply:
+1

https://internal-ops-us-east-0.grafana.net/design-system/Components/button/#donts

Reply:
👍